rand_distr/src/binomial.rs

```diff
@@ -50,10 +46,34 @@ use rand::Rng;
 #[derive(Clone, Copy, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Binomial {
-    /// Number of trials.
+    method: Method,
+    flipped: bool,
+    n: u64,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+enum Method {
+    Binv(Binv),
+    Btpe(Btpe),
+    Poisson(crate::poisson::KnuthMethod<f64>),
+    Constant(u64),
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+struct Binv {
+    r: f64,
+    s: f64,
+    a: f64,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+struct Btpe {
     n: u64,
-    /// Probability of success.
     p: f64,
+    // TODO consider precomputing more constants, there is a size/speed tradeoff to make here.
```

Comment:
Note: the size tradeoff doesn't come into play until we match `Binv` for size, so we may as well add *something* here. Maybe `c`?

Edit: maybe better to remove `Binv::a` to drop the whole structure down to 32 bytes.

Reply:
Revisiting this:

- Dropping `Binv::a` has approx ~12% perf. loss in binomial (10, 0.9) and (100, 0.99) benchmarks, but only ~2% in (1, 0.9)
- Dropping `Btpe::p1` has approx 6% perf. loss
- Dropping `Btpe::npq` has no significant perf. loss
- Adding `Btpe::c` has approx 3% perf. improvement
- Adding `Btpe::f_m` has approx 2-5% perf. improvement
- Adding `Btpe::m` has approx 5-11% perf. improvement

Those int-float conversions are more expensive than I thought. My conclusion is that we should keep `Binv` as-is, but replace `npq` with `m` in `Btpe`. (Or we remove `Binv::a` and `Btpe::npq` for the size reduction over perf; unfortunately we can't really tell which is more important in general.)

Reply:
(100,0.99) should use btpe, so this seems a bit off